src/main/kotlin/org/onflow/flow/sdk/models.kt

```diff
@@ -74,24 +74,22 @@ enum class SignatureAlgorithm(
 enum class HashAlgorithm(
     val algorithm: String,
     val outputSize: Int,
-    val id: String,
     val code: Int,
     val index: Int
 ) {
-    UNKNOWN("unknown", -1, "unknown", -1, 0),
-    SHA2_256("SHA-256", 256, "SHA256withECDSA", 1, 1),
-    SHA2_384("SHA-384", 384, "SHA384withECDSA", 1, 2),
-    SHA3_256("SHA3-256", 256, "SHA3-256withECDSA", 3, 3),
-    SHA3_384("SHA3-384", 384, "SHA3-384withECDSA", 3, 4);
+    UNKNOWN("unknown", -1, -1, 0),
+    SHA2_256("SHA-256", 256, 1, 1),
+    SHA3_256("SHA3-256", 256, 3, 2),
+    KMAC128("KMAC128", 128, 5, 3),
+    KECCAK256("KECCAK256", 256, 6, 4);
+
 
     companion object {
         @JvmStatic
-        fun fromCode(code: Int): HashAlgorithm = entries
-            .find { it.code == code } ?: UNKNOWN
+        fun fromCode(code: Int): HashAlgorithm = entries.find { it.code == code } ?: UNKNOWN
 
         @JvmStatic
-        fun fromCadenceIndex(index: Int): HashAlgorithm = entries
-            .find { it.index == index } ?: UNKNOWN
+        fun fromCadenceIndex(index: Int): HashAlgorithm = entries.find { it.index == index } ?: UNKNOWN
```

Comment:
Looking at this, it seems that `index` is interpreted as the Cadence index from https://cadence-lang.org/docs/language/crypto. So I think the hash indices above are not right (the last field should be `index`, not the last last field)